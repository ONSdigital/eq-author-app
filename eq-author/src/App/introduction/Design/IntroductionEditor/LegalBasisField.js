import React from "react";
import PropTypes from "prop-types";

import { Input } from "components/Forms";
import {
  RadioLabel,
  RadioField,
  RadioTitle,
  RadioDescription,
} from "components/Radio";

export const LegalOption = ({ name, value, children, onChange, selected }) => (
  <RadioLabel htmlFor={value} selected={selected}>
    <Input
      type="radio"
      variant="radioBox"
      checked={selected}
      name={name}
      id={value}
      value={value}
      onChange={onChange}
    />
    {children}
  </RadioLabel>
);
LegalOption.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onChange: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired,
};

const OPTIONS = [
  {
    title: "Notice 1",
    notice: true,
    description:
      "Notice is given under section 1 of the Statistics of Trade Act 1947.",
    value: "NOTICE_1",
  },
  {
    title: "Notice 2",
    notice: true,
    description:
      "Notice is given under sections 3 and 4 of the Statistics of Trade Act 1947.",
    value: "NOTICE_2",
  },
  {
    title: "Voluntary",
    description: "No legal notice will be displayed.",
    value: "VOLUNTARY",
  },
];

const LegalBasisField = ({ name, value, onChange, ...rest }) => (
  <RadioField {...rest}>
    {OPTIONS.map((option) => (
      <LegalOption
        key={option.value}
        name={name}
        value={option.value}
        selected={option.value === value}
        onChange={onChange}
      >
        <RadioTitle>
          {`${option.title} ${
            option.notice ? "- Your response is legally required." : ""
          }`}
        </RadioTitle>
        <RadioDescription>{option.description}</RadioDescription>
      </LegalOption>
    ))}
  </RadioField>
);

LegalBasisField.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default LegalBasisField;
