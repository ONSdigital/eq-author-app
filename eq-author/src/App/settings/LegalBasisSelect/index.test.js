import React from "react";
import { shallow } from "enzyme";

import LegalBasisField from ".";

const mockUseUpdateTheme = jest.fn();

jest.mock("hooks/useUpdateTheme", () => ({
  __esModule: true,
  default: jest.fn(() => mockUseUpdateTheme),
}));

describe("LegalBasisField", () => {
  let props;

  beforeEach(() => {
    props = {
      legalBasis: "NOTICE_1",
      shortName: "default",
      questionnaireId: "my-fave-questionnaire",
    };
  });

  it("should render", () => {
    expect(shallow(<LegalBasisField {...props} />)).toMatchSnapshot();
  });

  it("should show the option matching the value as selected", () => {
    const option = shallow(<LegalBasisField {...props} />).find(
      "[selected=true]"
    );
    expect(option.prop("value")).toEqual(props.legalBasis);
  });

  it("picking the option should update the theme's legal basis", () => {
    shallow(<LegalBasisField {...props} />)
      .find("[value='NOTICE_2']")
      .simulate("change", { name: props.name, value: "NOTICE_2" });

    expect(mockUseUpdateTheme).toHaveBeenCalledWith({
      questionnaireId: props.questionnaireId,
      shortName: props.shortName,
      legalBasisCode: "NOTICE_2",
    });
  });
});
