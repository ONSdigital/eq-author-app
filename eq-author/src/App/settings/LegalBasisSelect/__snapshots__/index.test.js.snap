// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegalBasisField should render 1`] = `
<Radio__RadioField>
  <LegalBasisSelect__StyledLabel>
    Legal basis
  </LegalBasisSelect__StyledLabel>
  <p>
    The legal basis appears on the survey introduction page.
  </p>
  <LegalOption
    key="NOTICE_1"
    name="default"
    onChange={[Function]}
    selected={true}
    value="NOTICE_1"
  >
    <Radio__RadioDescription>
      Section 1 of the Statistics of Trade Act 1947.
    </Radio__RadioDescription>
  </LegalOption>
  <LegalOption
    key="NOTICE_2"
    name="default"
    onChange={[Function]}
    selected={false}
    value="NOTICE_2"
  >
    <Radio__RadioDescription>
      Sections 2 and 3 of the Statistics of Trade Act 1947.
    </Radio__RadioDescription>
  </LegalOption>
  <LegalOption
    key="NOTICE_3"
    name="default"
    onChange={[Function]}
    selected={false}
    value="NOTICE_3"
  >
    <Radio__RadioDescription>
      Sections 3 and 4 of the Statistics of Trade Act 1947.
    </Radio__RadioDescription>
  </LegalOption>
  <LegalOption
    key="NOTICE_NI"
    name="default"
    onChange={[Function]}
    selected={false}
    value="NOTICE_NI"
  >
    <Radio__RadioDescription>
      Article 5 of the Statistics of Trade and Employment (Northern Ireland) Order 1988.
    </Radio__RadioDescription>
  </LegalOption>
  <LegalOption
    key="NOTICE_FUELS"
    name="default"
    onChange={[Function]}
    selected={false}
    value="NOTICE_FUELS"
  >
    <Radio__RadioDescription>
      Petroleum Act 1998 and Section 1 of the Statistics of Trade Act 1947.
    </Radio__RadioDescription>
  </LegalOption>
  <LegalOption
    key="VOLUNTARY"
    name="default"
    onChange={[Function]}
    selected={false}
    value="VOLUNTARY"
  >
    <Radio__RadioDescription>
      Voluntary
    </Radio__RadioDescription>
  </LegalOption>
</Radio__RadioField>
`;
